gen/build_deploy/bash.py

```diff
@@ -669,24 +668,36 @@ def fill_template(base_name, format_args):
     return installer_filename
 
 
-def make_dcos_launch():
-    # NOTE: this needs to be kept in sync with build_dcos_launch.sh
+def get_launch_package_id(package_list):
+    """scans througha the package list of the
+    """
+    launch_package_id = None
+    for package_id in package_list:
+        if pkgpanda.PackageId(package_id).name == 'dcos-launch':
+            launch_package_id = package_id
+            break
+    assert launch_package_id, 'dcos-launch package cannot be found!'
+    return launch_package_id
+
+
+def fetch_dcos_launch_bin(launch_name, launch_package_id):
     work_dir = py.path.local.mkdtemp()
-    # the tree will not be in the working dir if run from downstream
-    tree_src = './'
-    if os.path.exists('ext/upstream'):
-        tree_src = './ext/upstream'
-    src_info = {'kind': 'git_local', 'rel_path': tree_src}
-    this_tree = GitLocalSrcFetcher(src_info, None, os.getcwd())
-    this_tree.checkout_to(str(work_dir))
-    # put the launch spec into the root of the tree before running pyinstaller
-    work_dir.join('dcos-launch.spec').write(pkg_resources.resource_string('launch', 'dcos-launch.spec'))
+    # inspect the local packages cache to grab the desired artifact
+    cached_package_path = 'packages/cache/packages/dcos-launch/' + launch_package_id + '.tar.xz'
+    temp_package_path = str(work_dir.join('dcos-launch.tar.xz'))
+    subprocess.check_call(['cp', cached_package_path, temp_package_path])
     with work_dir.as_cwd():
-        subprocess.check_call(['pyinstaller', '--log-level=DEBUG', 'dcos-launch.spec'])
-    subprocess.check_call(['mv', str(work_dir.join('dist').join('dcos-launch')), 'dcos-launch'])
-    work_dir.remove()
+        subprocess.check_call(['tar', '-Jxf', 'dcos-launch.tar.xz'])
+    launch_path = 'packages/cache/' + launch_name
+    subprocess.check_call(['cp', str(work_dir.join('dcos-launch')), launch_path])
+    return launch_path
+
 
-    return "dcos-launch"
+def get_dcos_launch_name(variant):
+    if variant is None:
+        return 'dcos-launch'
+    else:
+        return 'dcos-launch-' + pkgpanda.util.variant_name(variant)
```

Comment:
Could we replace this with something in `pkgpanda.util`? It looks similar to what `variant_prefix()` is doing. Maybe add a `def variant_suffix(variant, delim='.')` and use `launch_name = 'dcos-launch' + variant_suffix(variant, delim='-')`.